src/vs/workbench/contrib/terminalContrib/links/browser/terminalLinkDetectorAdapter.ts

```diff
@@ -90,7 +97,10 @@ export class TerminalLinkDetectorAdapter extends Disposable implements ILinkProv
 			endLine++;
 		}
 
-		const detectedLinks = await this._detector.detect(lines, startLine, endLine);
+		const detectResult = this._detector.detect(lines, startLine, endLine);
+		const detectedLinks = Array.isArray(detectResult) ? detectResult : (await raceTimeout(detectResult, Constants.DetectTimeBudget, () => {
```

Comment:
It looks like the API in .d.ts supports cancellation tokens. Should the token be cancelled on timeout here? Could avoid queuing a ton of really slow link-providing operations